apis/v1alpha2/validation/validation.go

```diff
@@ -34,6 +36,23 @@ var (
 	}
 )
 
+// ValidateGatewayClassName validates that the given name can be used as
+// a GatewayClass name.
+var ValidateGatewayClassName = apivalidation.NameIsDNSSubdomain
+
+// ValidateGatewayClass validates gc according to the Gateway API specification.
+// For additional details of the GatewayClass spec, refer to:
+// https://gateway-api.sigs.k8s.io/spec/#networking.x-k8s.io/v1alpha2.GatewayClass
+func ValidateGatewayClass(gc *gatewayv1a2.GatewayClass) field.ErrorList {
+	return validateGatewayClassObjMeta(&gc.ObjectMeta, field.NewPath("metadata"))
+}
+
+// validateGatewayClassObjMeta validates whether required fields of metadata are set according
+// to the Gateway API specification.
+func validateGatewayClassObjMeta(meta *metav1.ObjectMeta, path *field.Path) field.ErrorList {
+	return apivalidation.ValidateObjectMeta(meta, false, ValidateGatewayClassName, path)
```

Comment:
suggest we just inline this call

Reply:
Is the API-server not already doing this kind of validation for custom resources?

Reply:
I'm not exactly sure how this works, but it does appear that CRD validation already exists. Simply applying a GatewayClass with an invalid name without this validation gives the following error message:

```
The GatewayClass "aAcme-lb" is invalid: metadata.name: Invalid value: "aAcme-lb": a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')
```

The [main validation function](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/validation/validation.go#L47) does not appear to include this but it must happen further up the stack.